### PR TITLE
feat: add support for not-in and not-eq query operators

### LINKED ISSUE #189: Add support for != and not-in queries
Porting from [Java](https://github.com/googleapis/java-firestore/pull/350) and [Node](https://github.com/googleapis/nodejs-firestore/pull/1292)

For python auto generated code is missing here  whereas it's generated for java.
https://github.com/googleapis/python-firestore/blob/581520c67340680927148045249bd775487eabac/google/cloud/firestore_v1/types/query.py#L173-L183

@tseaver @crwilcox  Should i run synth.py for that?

### PR DESCRIPTION
**Pull request by HemangChothani**
Fixes #189

### CODE REVIEW

File: google/cloud/firestore_v1/base_query.py [hunk 1]
```diff
@@ -847,7 +849,7 @@ def _enum_from_op_string(op_string) -> Any:
 
     Args:
         op_string (str): A comparison operation in the form of a string.
-            Acceptable values are ``<``, ``<=``, ``==``, ``>=``
+            Acceptable values are ``<``, ``<=``, ``==``, ``!=``, ``>=``
```
Comment (crwilcox) [inline google/cloud/firestore_v1/base_query.py:852]: does this method also accept "not-in, array_contains" and more?

**Review by crwilcox — APPROVED** [review]

File: google/cloud/firestore_v1/base_query.py (hunk 1 above)
Comment (HemangChothani) [reply google/cloud/firestore_v1/base_query.py:852]: @crwilcox  Yes, this method accept all the values  including `array_contains` we don't have any special condition for that, limitation of acceptable values are just mentioned in doc-string.
Is this question related to https://github.com/googleapis/python-firestore/issues/203 issue?

**Comment by tseaver** [comment]
@HemangChothani Please fix the [conformance test failure](https://source.cloud.google.com/results/invocations/c95b2464-c58e-4c5d-b9f6-0a767b7a226c/targets) on this branch.

File: tests/unit/v1/testdata/query-invalid-operator.json [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-              "op": "!=",
-              "jsonValue": "4"
-            }
-          }
-        ],
-        "isError": true
-      }
-    }
-  ]
-}
```
Comment (tseaver) [inline tests/unit/v1/testdata/query-invalid-operator.json:25]: The [upstream for this file](https://github.com/googleapis/conformance-tests/blob/master/firestore/v1/query-invalid-operator.json) is authoritative:  we are not supposed to modify / remove the copied-in files.

@BenWhitehead please clarify the intention here.

Comment (BenWhitehead) [reply tests/unit/v1/testdata/query-invalid-operator.json:25]: PR to update the operator in the upstream files: https://github.com/googleapis/conformance-tests/pull/41

Comment (BenWhitehead) [reply tests/unit/v1/testdata/query-invalid-operator.json:25]: Above PR has been merged and can be pulled into the project to test the actual case for a not supported operator.

**Review by tseaver — CHANGES_REQUESTED** [review]
@HemangChothani Please revert commit 420788bcc89b22737a63e70d612d6c32414d7304 and copy in the version of `query-invalid-operator.json` updated by https://github.com/googleapis/conformance-tests/pull/41/.

**Review by tseaver — APPROVED** [review]

**Pull request merged**